mcs/class/corlib/System.Runtime.Remoting/RemotingServices.cs

```diff
@@ -808,45 +808,14 @@ internal static object DeserializeCallData (byte[] array)
 		[SecurityPermission (SecurityAction.Assert, SerializationFormatter = true)] // FIXME: to be reviewed
 		internal static byte[] SerializeExceptionData (Exception ex)
 		{
+			byte[] result = null;
 			try {
-				int retry = 4;
-				
-				do {
-					try {
-						MemoryStream ms = new MemoryStream ();
-						_serializationFormatter.Serialize (ms, ex);
-						return ms.ToArray ();
-					}
-					catch (Exception e) {
-						if (e is ThreadAbortException) {
-#if MONO_FEATURE_THREAD_ABORT
-							Thread.ResetAbort ();
-#endif
-							retry = 5;
-							ex = e;
-						}
-						else if (retry == 2) {
-							ex = new Exception ();
-							ex.SetMessage (e.Message);
-							ex.SetStackTrace (e.StackTrace);
-						}
-						else
-							ex = e;
-					}
-					retry--;
-				}
-				while (retry > 0);
-				
-				return null;
-			}
-			catch (Exception tex)
-			{
-				byte[] data = SerializeExceptionData (tex);
-#if MONO_FEATURE_THREAD_ABORT
-				Thread.ResetAbort ();
-#endif
-				return data;
+			} finally {
```

Comment:
This could really do with a comment... `try { /* empty */ } finally { ... }` is non-obvious